lint/settings.py

```diff
@@ -86,18 +85,27 @@ def validate_settings():
     schema_file = "resources/settings-schema.json"
     schema = util.load_json(schema_file, from_sl_dir=True)
     window = sublime.active_window()
-    util.clear_message()
     good = True
 
     for name, settings in get_settings_objects():
         if settings:
             try:
-                validate(settings, schema)
+                validate(settings, schema, format_checker=FormatChecker())
             except ValidationError as error:
                 good = False
-                error_msg = error.message.split("\n")[0]  # reduce verbosity
-                full_msg = "Invalid settings in '{}':\n{}".format(name, error_msg)
-
-                logger.error(full_msg)
+                path_to_err = ('->'.join(
```

Comment:
I'd prefer ` > ` to join with visually, or `.` because that's JS syntax.

Reply:
`.`s are out bc our keys can contain points or dashes. So the JSON is not a JS object.

I merge with `>` but if you can come up with a really good formatting, just write it down. For now, I still drop array indices so it print `styles>icon` but styles is a list and we don't tell the user which style definition failed. (Although the validator knows.) But `styles > 0 > icon` seemed ugly.

Reply:
Hm, if it’s not a JS (-like) path (which would be my ideal), it’s good like this. Actually, we should use a different character than `&gt;`, I’ll look it up later.